src/query/service/src/pipelines/processors/transforms/hash_join/hash_join_state_impl.rs

```diff
@@ -137,7 +137,7 @@ impl HashJoinState for JoinHashTable {
                     .map(|x| Some(*x))
                     .collect(),
                 JoinType::RightMark => {
-                    if !has_null {
+                    if !has_null && !chunk.cols.is_empty() {
```

Comment:
Bug in right mark join: `cols` is possibly empty; Such as the  case: ` select 1 = all(select 1);` cc @youngsofun

Reply:
ok